instrumentation/apache-camel-2.20/javaagent/src/test/groovy/test/SqsCamelTest.groovy

```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package test
+
+import static io.opentelemetry.api.trace.Span.Kind.CLIENT
+import static io.opentelemetry.api.trace.Span.Kind.CONSUMER
+import static io.opentelemetry.api.trace.Span.Kind.INTERNAL
+import static io.opentelemetry.api.trace.Span.Kind.PRODUCER
+
+import com.google.common.collect.ImmutableMap
+import io.opentelemetry.instrumentation.test.AgentTestRunner
+import io.opentelemetry.instrumentation.test.utils.PortUtils
+import org.apache.camel.CamelContext
+import org.apache.camel.ProducerTemplate
+import org.elasticmq.rest.sqs.SQSRestServer
+import org.elasticmq.rest.sqs.SQSRestServerBuilder
+import org.springframework.boot.SpringApplication
+import org.springframework.context.ConfigurableApplicationContext
+import spock.lang.Shared
+
+class SqsCamelTest extends AgentTestRunner {
+
+  @Shared
+  ConfigurableApplicationContext server
+
+  @Shared
+  SQSRestServer sqs
```

Comment:
Nit: have you considered running localstack testcontainers instead: https://www.testcontainers.org/modules/localstack/ ?

Reply:
Localstack uses elasticMQ underneath :D

Reply:
As long as that small use case doesn't cause flakiness anything's fine :) We have just had better success with test containers I think.